### PR TITLE
Add readonly mode

### PR DESCRIPTION
**Pull request by yuxitang-amzn**
*Issue #, if available:*

*Description of changes:*
Add readonly mode that can only see dashboards (configured by `readonly_modes.roles`, default to "kibana_read_only").

Known defects:
This PR didn't disable home app, on home app there are a number of links (e.g. dashboards, visualizations), click thru them will see an "Application not found error". But we didn't find a way to change default app to dashboards, means if we disable home app, after login user will be redirect to home app and see "Application not found error".

Screenshots:
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/types.ts [hunk 1]
```diff
@@ -32,6 +32,9 @@ export interface AuthInfo {
 }
 
 export interface ClientConfigType {
+  readonly_mode: {
+    roles: string[];
```
Comment (zengyan-amazon) [inline public/types.ts:36]: you may also want to export this config in `server/index.ts`

Comment (yuxitang-amzn) [reply public/types.ts:36]: It's already there.
https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/96ba75a71e75490dc41f89e9d3f8453893eb7a77/server/index.ts#L24

Comment (zengyan-amazon) [reply public/types.ts:36]: but you need to make it available by adding it to exportToBrowser https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/96ba75a71e75490dc41f89e9d3f8453893eb7a77/server/index.ts#L189

Comment (yuxitang-amzn) [reply public/types.ts:36]: Got it, fixed.

**Review by zengyan-amazon — APPROVED** [review]

**Pull request merged**